param/parameterized.py

```diff
@@ -437,10 +431,15 @@ class Foo(Bar):
                  'pickle_default_value','allow_None',
                  'subscribers','_owner']
 
-    # When created, a Parameter does not know which
-    # Parameterized class owns it. If a Parameter subclass needs
-    # to know the owning class, it can declare an 'objtype' slot
-    # (which will be filled in by ParameterizedMetaclass)
+    # Note: When initially created, a Parameter does not know which
+    # Parameterized class owns it, nor does it know its names
+    # (attribute name, internal name). Once the owning Parmaeterized
+    # class is created, _owner, _attrib_name, and _internal name are
+    # set.
+
+    # TODO regarding _attrib_name, _owner: what if someone re-uses
+    # a parameter object across different classes? we should raise
+    # an error if attrib name,owner already set
```

Comment:
Raising an error sounds good, and the message could tell people how to avoid the message (by cloning the Parameter object somehow before using it in a new class.  

I'm trying to figure out how to make it reasonable to add an _owner_instance slot, but it ties my brain in knots.  Basically, I want to be able to ask a Parameterized instance for an object that has all the information in the Parameter, plus a pointer to the actual instance so that the value can be accessed.  Yet there's only one Parameter object per class, not per instance, so I guess it would have to be a copy of the Parameter object?  And then it wouldn't be the one that actually applies to this object.  So I guess that before having _owner_instance, we'd first have to set up allowing a local instance copy of the Parameter to override the class's Parameter, at which point the real Parameter just for this instance could be returned with _owner_instance filled in?